src/uct/api/uct.h

```diff
@@ -1213,6 +1214,55 @@ struct uct_md_attr {
 };
 
 
+/**
+ * @ingroup UCT_MD
+ * @brief UCT MD memory attributes field mask
+ *
+ * The enumeration allows specifying which fields in @ref uct_md_mem_attr_t
+ * are present.
+ */
+enum uct_md_mem_attr_field {
+    UCT_MD_MEM_ATTR_MEM_TYPE = UCS_BIT(0), /**< Indicate if memory type is populated. Types
+                                                belong to one of @ref ucs_memory_type_t */
+    UCT_MD_MEM_ATTR_SYS_DEV  = UCS_BIT(1)  /**< Indicate if details of system device backing
+					        the pointer is populated. @ref ucs_sys_device_t*/
+};
+
+
+/**
+ * @ingroup UCT_MD
+ * @brief  Memory domain attributes.
+ *
+ * This structure defines the attributes of a memory pointer which may
+ * include memory type of the pointer, the system device that backs the
+ * pointer depending on the bit fields populated in field_mask
+ */
+typedef struct uct_md_mem_attr {
+    uint64_t          field_mask; /**< @ref uct_md_mem_attr_t */
```

Comment:
pls expand a bit more on the usage of field_mask ,see https://github.com/openucx/ucx/blob/master/src/uct/api/uct.h#L1142 for example